test/unit/modules/blocks/chain.js

```diff
@@ -0,0 +1,636 @@
+'use strict';
+
+describe('blocks/chain', function () {
+
+	describe('Chain constructor', function () {
+
+		describe('library', function () {
+
+			it('should assign logger');
+
+			it('should assign db');
+
+			it('should assign genesisblock');
+
+			it('should assign bus');
+
+			it('should assign balanceSequence');
+
+			describe('should assign logic',function () {
+
+				it('should assign block');
+
+				it('should assign transaction');
+			});
+		});
+
+		it('should set self to this');
+
+		it('should call library.logger.trace"');
+		
+		it('should call library.logger.trace with "Blocks->Chain: Submodule initialized."');
+
+		it('should return self');
+	});
+
+	describe('saveGenesisBlock', function () {
+
+		it('should call library.db.query');
+
+		it('should call library.db.query with sql.getGenesisBlockId');
+
+		it('should call library.db.query with { id: library.genesisblock.block.id }');
+
+		describe('when db query fails', function () {
+
+			it('should call logger.error');
+			
+			it('should call logger.error with err.stack');
+
+			it('should call callback with "Blocks#saveGenesisBlock error"');
+		});
+
+		describe('when db query succeeds', function () {
+
+			describe('and blockId is undefined', function () {
+
+				it('should call self.saveBlock');
+
+				it('should call self.saveBlock with library.genesisblock.block');
+				
+				describe('when self.saveBlock fails', function () {
+					
+					it('should call callback with error');
+				});
+				
+				describe('when self.saveBlock succeeds', function () {
```

Comment:
the block not needed - the same action will take place also for if block id will be present